contrib/monitored_resource_util/build.gradle

```diff
@@ -0,0 +1,17 @@
+description = 'OpenCensus Monitored Resource Util'
+
+apply plugin: 'java'
+
+[compileJava, compileTestJava].each() {
+    it.sourceCompatibility = 1.6
+    it.targetCompatibility = 1.6
+}
+
+dependencies {
+    compileOnly libraries.auto_value
+
+    compile libraries.google_cloud_core,
+            libraries.guava
```

Comment:
Based on https://github.com/census-instrumentation/opencensus-java/issues/1113 I think we should avoid this. I see few usages (especially using collect you can use https://www.tutorialspoint.com/java/util/collections_unmodifiablelist.htm).

Can we please limit the usage of guava?